service/s3control/internal/customizations/process_arn_resource.go

```diff
@@ -273,7 +273,8 @@ func buildOutpostAccessPointRequest(ctx context.Context, options outpostAccessPo
 	// resolve regional endpoint for resolved region.
 	var endpoint aws.Endpoint
 	var err error
-	if endpointsID == "s3" {
+	endpointSource := awsmiddleware.GetEndpointSource(ctx)
+	if endpointsID == "s3" && endpointSource != aws.EndpointSourceCustom {
```

Comment:
For future compatibility should this be `endpointSource == aws.EndpointSourceServiceMetadata` Since potential future additions to `EndpointSource` would trigger this behavior as well. Potentially unexpectedly.

Reply:
Since these are customizations, future endpoint source additions should always trigger the behavior unless explicitly stated. The customization doesn't add labels only if it is a custom endpoint source. Everything else should be allowed IMO.

Reply:
Will take that back, what other endpoint source types can there be?

Reply:
Reason for suggesting this is we don't know what kind of endpoint sources will be defined in the future. Its safer to state only perform this behavior for the endpoint source we know it should be applied to today. Then tomorrow when a new feature or endpoint source is defined we can safely add that capability as needed.